src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -457,6 +472,15 @@ private async ValueTask ProcessHeadersFrame(FrameHeader frameHeader)
             while (!frameHeader.EndHeadersFlag)
             {
                 frameHeader = await ReadFrameAsync().ConfigureAwait(false);
+
+                // We may receive ping frames beetween the HEADERS/CONTINUATION frames.
+                // These are typically a PING ACKs in response to RTT PINGs.
+                if (frameHeader.Type == FrameType.Ping)
+                {
+                    ProcessPingFrame(frameHeader);
+                    continue;
+                }
```

Comment:
I changed this to fix a test failure, and it seems very reasonable to me that PING ACKs may appear here, considering they have priority. However, according to  [to section 6.10](https://datatracker.ietf.org/doc/html/rfc7540#section-6.10):

>   A CONTINUATION frame MUST be preceded by a HEADERS, PUSH_PROMISE or CONTINUATION frame without the END_HEADERS flag set.

Shall we undo this, or does it make sense to be permissive?

Reply:
Are you seeing an actual server do this, or is it just our test?

I don't think we should allow this unless we know for sure that some real server we care about is doing this.

Reply:
I've seen it in our test server only, don't know how would a real server react, though if they follow RFC literally, the following will happen:

1. Client receives HEADERS
2. Client sends a PING
3. Server sends a lot of header data in a series of CONTINUATION frames
4. Server sends PING ACK in response to 2.

There will be significant turnaround delay for the PING-PONG, which doesn't make sense to me.

/cc @halter73 any thoughts?

Reply:
> as long as the preceding frame is **on the same
   stream** and is a HEADERS, PUSH_PROMISE, or CONTINUATION frame without
   the END_HEADERS flag set.

AFAIK PING/PONGs do not belong to any particular stream, thus interleaving them with DATA/HEADERS frames is OK I think.

Reply:
> Header blocks
   MUST be transmitted as a contiguous sequence of frames, with no
   interleaved frames of any other type or from any other stream.

https://datatracker.ietf.org/doc/html/rfc7540#section-4.3

Reply:
I see, thanks for pointing it out! Then yes, we shouldn't need to handle PING/PONGs in the middle of HEADER/CONTINUATIONs frame sequence.

Reply:
I gonna revert this, but the RFC doesn't make sense here IMO, PING should be an exception in section 4.3. + There is a nonzero chance that we will break some server that is not RFC-conformant.